cpp/src/arrow/compute/kernels/vector_sort_benchmark.cc

```diff
@@ -99,6 +99,40 @@ static void ChunkedArraySortFuncInt64Benchmark(benchmark::State& state,
   ArraySortFuncBenchmark(state, runner, std::make_shared<ChunkedArray>(chunks));
 }
 
+template <typename Runner>
+static void ChunkedArraySortFuncStringBenchmark(benchmark::State& state,
+                                                const Runner& runner, int32_t min_length,
+                                                int32_t max_length) {
+  RegressionArgs args(state);
+
+  // The array_size is derived from these two equations:
+  //
+  //     size = array_size * (1.0 - args.null_proportion) * string_mean_length
+  //     string_mean_length = (max_length + min_length) / 2.0
+  const double valid_proportion = 1.0 - args.null_proportion;
+  const double array_size =
+      (valid_proportion > std::numeric_limits<double>::epsilon())
+          ? (args.size * 2) / (valid_proportion * (max_length + min_length))
+          : (args.size * 2) / (max_length + min_length);
+
+  const auto n_chunks = 10;
+  const auto array_chunk_size = static_cast<int64_t>(std::round(array_size / n_chunks));
+  auto rand = random::RandomArrayGenerator(kSeed);
+
+  ArrayVector chunks;
+  int64_t generated_array_size_in_bytes = 0;
+  for (auto i = 0; i < n_chunks; ++i) {
+    auto values =
+        rand.String(array_chunk_size, min_length, max_length, args.null_proportion);
+    auto array = arrow::internal::checked_pointer_cast<const arrow::StringArray>(values);
+    generated_array_size_in_bytes += array->value_data()->size();
+    chunks.push_back(std::move(values));
+  }
+
+  state.counters["generated_array_size"] = generated_array_size_in_bytes;
```

Comment:
I'm not certain this is useful as a counter.

I would much rather see (for all of the benchmarks in this file and, ideally, all compute benchmarks) a bytes-per-second counter.  If I had that then I wouldn't care about generated_array_size.

Reply:
I accidentally pushed the commit that adds it here. I only wanted it temporarily to double-check my estimation formula.

Reply:
It will be gone next time I push.